View Post UI Skeleton

abbyhofmann opened the pull request:
### Description:
This PR contains the work for the UI skeleton of viewing a post. The following has been implemented:
- clicking on a post in the sidebar highlights the post and routes to the URL of /post/:pid
- skeleton of the individual response components (student answer, new student answer, instructor answer, new instructor answer, new followup discussion)
- added a 'hooks' folder

The following is left to-do:
- remove the hardcoding (which will involve adding the endpoints for fetching a post by post id, then passing that post's student answer/instructor answer/followup discussion data to their corresponding components to be displayed)
- followup discussions UI has not been started at all
- anywhere there's a "TODO" comment lol

Video of the functionality - I switched the components being displayed by updating the JSON object it is getting the data from:

https://github.com/user-attachments/assets/656bf3fb-9392-4acd-83f9-2a3342adc3b8

Commit:
post item component extraction

Commit:
formatting

Commit:
wip

Commit:
view post page wip

Commit:
working on nav and state of selected post

Commit:
wip

Commit:
post box updates

Commit:
selected post turns yellow in sidebar

Commit:
additional components but still hard coded

Commit:
usepostsidebar hook

Commit:
comments

Commit:
Merge branch 'main' into view-post

sanjana-kashyap commented:
i'm not sure if it's my laptop, but i can't see the post content next to the list of questions - i can only see it below the questions where the `Class at a Glance` section used to be (which is also at the bottom for me).

https://github.com/user-attachments/assets/56e39f10-be89-481d-bb0f-2b38a2cddc4a

sanjana-kashyap commented:
i think visually, everything looks great! would love to see what an instructor post looks like but that probably will show up after removing the hard-coded content i'm assuming. great job!!!!

sanjana-kashyap commented on src/Kambaz/Courses/Piazza/ViewPost/PostBox.tsx:
```diff
@@ -0,0 +1,75 @@
+
+    const { post } = props;
+
+    console.log('post: ', post)
```
might wanna remove this log

abbyhofmann commented:
> i think visually, everything looks great! would love to see what an instructor post looks like but that probably will show up after removing the hard-coded content i'm assuming. great job!!!!

Ooooh yeah I can show that

abbyhofmann commented:
> i'm not sure if it's my laptop, but i can't see the post content next to the list of questions - i can only see it below the questions where the `Class at a Glance` section used to be (which is also at the bottom for me).
> 
>  Screen.Recording.2025-03-13.at.11.25.01.mov

hmmm let me look into this

abbyhofmann commented:
> > i'm not sure if it's my laptop, but i can't see the post content next to the list of questions - i can only see it below the questions where the `Class at a Glance` section used to be (which is also at the bottom for me).
> > Screen.Recording.2025-03-13.at.11.25.01.mov
> 
> hmmm let me look into this

ok yeah it's also at the bottom for me... I can make it so that the "view post" and "class at a glance" pages appear next to the sidebar

sanjana-kashyap commented:
also a minor problem, but the piazza logo doesn't show up for me and results in an error in the console because of a failed GET request, don't know if that's a me-problem? 🤔

Commit:
need to remove page scrollbar

Commit:
created folders, code review, fixed alignment to sidebar

Commit:
import fix; still need to remove overall scrollbar on main page

Commit:
piazza logo added for navbar

abbyhofmann commented:
https://github.com/user-attachments/assets/d1933277-e8d3-4cf3-af36-108198b1d4eb

Updated video with the following changes: 
- added the piazza logo in the nav bar (it's white so you can't see it) 
- made the pages display to the right of the sidebar 
- added a scrollbar to the sidebar and the view post page (need to remove the outer scrollbar on the page overall but I'm having trouble doing so)

Commit:
comments removed
